frontend/src/component/common/AutocompleteBox/AutocompleteBox.tsx

```diff
@@ -9,13 +8,16 @@ import {
 } from '@mui/material';
 import type { AutocompleteRenderInputParams } from '@mui/material/Autocomplete';
 import { useState } from 'react';
+import type { ReactNode } from 'react';
 
 interface IAutocompleteBoxProps {
     label: string;
     options: IAutocompleteBoxOption[];
     value?: IAutocompleteBoxOption[];
     onChange: (value: IAutocompleteBoxOption[]) => void;
     disabled?: boolean;
+    icon?: ReactNode | null;
```

Comment:
Can we make it just normal or undefined? I don't think the `null` state adds anything here (except for possibly confusion about what the difference between null and undefined is).

```suggestion
    icon?: ReactNode;
```

Reply:
On closer inspection, I see you treat them differently: `undefined` is the default icon, `null` is no icon. Maybe that is sensible? 🤔